mozci/push.py

```diff
@@ -343,6 +367,68 @@ def add(result):
 
         return [Task.create(**task) for task in normalized_tasks]
 
+    def gather_missing_results(self, tasks: list) -> list:
+        """ Fetch and cache errors and results from Taskcluster. Cache together AD and TC results."""
+        test_tasks = {}  # Structure to be cached
+        tasks_missing_results = []  # List of tasks without data from AD
+
+        # Let's gather all collected data from AD
+        for t in tasks:
+            if isinstance(t, TestTask):
+                if not (t._results is None):
+                    # Data for this task has been gathered
+                    test_tasks[t.id] = {
+                        "errors": t._errors,
+                        "results": t._results,
+                    }
+                else:
+                    tasks_missing_results.append(t)
```

Comment:
You should be able to remove all of this code and just get errors and results for all tasks. If the data is available already, it will be returned, if the data is not available already, it will be gathered from errorsummary.log. There should be no need to first get the tasks with missing results.

Reply:
Cool! That makes sense. Changing it now.